### PR TITLE
Feat: resize PVC dynamically

### PR DESCRIPTION
**Pull request by PowerfooI**
## Summary

1. Support for resizing PVC dynamically if the StorageClass allow volume expansion #150 
2. Fix stuck tenants when memory size of unit is smaller than `__min_full_resource_pool_memory` #177 
3. Expose operator metrics through service #198

### CODE REVIEW

File: internal/resource/obcluster/utils.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package obcluster
+
+import "github.com/oceanbase/ob-operator/api/v1alpha1"
+
+func (m *OBClusterManager) doStorageSizeExpand(obzone *v1alpha1.OBZone) bool {
```
Comment (chris-sun-star) [inline internal/resource/obcluster/utils.go:5]: doStorageSizeExpand is a little bit confused, it feels like actually do the size expantion

File: internal/resource/obcluster/obcluster_task.go [hunk 2]
```diff
@@ -935,48 +935,57 @@ func (m *OBClusterManager) CreateServices() tasktypes.TaskError {
...
-			obzone.Spec.OBServerTemplate.Resource.Memory != m.OBCluster.Spec.OBServerTemplate.Resource.Memory {
-			obzone.Spec.OBServerTemplate.Resource.Cpu = m.OBCluster.Spec.OBServerTemplate.Resource.Cpu
-			obzone.Spec.OBServerTemplate.Resource.Memory = m.OBCluster.Spec.OBServerTemplate.Resource.Memory
-			err := m.Client.Update(m.Ctx, &obzone)
-			if err != nil {
-				return errors.Wrap(err, "update obzone")
-			}
-		}
-	}
+func (m *OBClusterManager) scaleUpOBZoneChanger(obzone *v1alpha1.OBZone) {
```
Comment (chris-sun-star) [inline internal/resource/obcluster/obcluster_task.go:940]: use a verb for function name

**Review by chris-sun-star — COMMENTED** [review]
1 PR for 1 ISSUE
consider to split the PR

**Comment by PowerfooI** [comment]
> 1 PR for 1 ISSUE consider to split the PR

OK, I'll do that. Some issues were quite small so I fixed them by the way.

**Review by chris-sun-star — APPROVED** [review]
LGTM

**Pull request merged**